tests/e2e/flowtest/test_base_operation_node_drain.py

```diff
@@ -0,0 +1,122 @@
+        )
+        logging.info("Started object IOs in background")
+
+        pvc_create_delete = executor_run_bg_ios_ops.submit(
+            bg_handler.handler,
+            create_pvc_delete,
+            multi_pvc_factory,
+            project,
+            iterations=70
+        )
+        logging.info("Started pvc create and delete in background")
+
+        pgsql_workload = executor_run_bg_ios_ops.submit(
+            bg_handler.handler,
+            pgsql_factory_fixture,
+            replicas=1, clients=1, transactions=100,
+            timeout=100, iterations=1
+        )
+        logging.info("Started pgsql workload in background")
+
+        flow_ops = FlowOperations()
+        logger.info("Starting operation 1: Node Drain")
+        node_name, nodes = flow_ops.node_operations_entry_criteria(
+            node_type='worker', number_of_nodes=1, operation_name="Node Drain"
+        )
+        # Node maintenance - to gracefully terminate all pods on the node
+        node.drain_nodes([node_name[0].name])
+        # Make the node schedulable again
+        node.schedule_nodes([node_name[0].name])
+        logger.info("Verifying exit criteria for operation 1: Node Drain")
+        flow_ops.validate_cluster(node_status=True, operation_name="Node Drain")
+
+        logger.info("Starting operation 2: Add Capacity")
+        osd_pods_before, restart_count_before = flow_ops.add_capacity_entry_criteria()
+        # Add capacity
+        osd_size = storage_cluster.get_osd_size()
+        result = storage_cluster.add_capacity(osd_size)
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        pod.wait_for_resource(
+            timeout=300,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=result * 3
+        )
+        logger.info("Verifying exit criteria for operation 2: Add Capacity")
+        flow_ops.add_capacity_exit_criteria(restart_count_before, osd_pods_before)
+
+        logger.info("Starting operation 3: Node Restart")
+        node_name, node_platform = flow_ops.node_operations_entry_criteria(
+            node_type='worker', number_of_nodes=1, operation_name="Node Restart"
+        )
+        # Node failure (reboot)
+        node_platform.restart_nodes(nodes=node_name)
+        logger.info("Verifying exit criteria for operation 3: Node Restart")
+        flow_ops.validate_cluster(node_status=True, operation_name="Node Restart")
+
+        logger.info("Starting operation 4: Node network fail")
+        node_name, node_platform, nw_fail_time = flow_ops.node_operations_entry_criteria(
+            node_type='worker', number_of_nodes=1, network_fail_time=300, operation_name="Node N/W failure"
+        )
+        # Node n/w interface failure
+        node.node_network_failure(node_name[0].name)
+        logger.info(f"Waiting for {nw_fail_time} seconds")
+        sleep(nw_fail_time)
```

Comment:
Do we need these?
In the node_network_failure(), when the network is failed on the node, we are waiting for it to reach not ready state.

Reply:
It is part of the test case which has two scenarios: 1. short down time after n/w failure 2. prolonged down time after n/w failure

Reply:
So decided to go with short wait time.